gtda/utils/validation.py

```diff
@@ -194,9 +195,9 @@ def _check_array_mod(X, **kwargs):
     accepted but infinity is."""
     if not kwargs['force_all_finite']:
         Xnew = check_array(X, **kwargs)
-        if np.isnan(Xnew).any():
+        if np.isnan(Xnew if not issparse(Xnew) else Xnew.data).any():
             raise ValueError(
-                "Input contains NaN. Only finite values and infinity are "
+                "Input contains NaNs. Only finite values and infinity are "
                 "allowed when parameter `force_all_finite` is False.")
         return Xnew
```

Comment:
It seems that the input is systematically copied  and returned. In `sklearn`, there is a ``copy`` boolean argument to the `check_array` function. This is especially useful if one of the function in `transform` is already returning a copy of the sample or if it does not modify the input at all.

Reply:
As far as I recall, this is not quite the case if the input is a 3D `ndarray` (do you agree?), because in that case we simply fall back to `sklearn` and in particular one can use the `copy` kwarg. In the list case, the situation is different and I'll look better into it.

Reply:
Yes, I agree.